core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -1435,28 +1485,89 @@ def get_minimum_resources(self, is_gpu_available=False) -> Dict[str, int]:
+            If expected memory usage is >max_memory_usage_error_ratio, raise NotEnoughMemoryError
+        mem_warning_threshold : float, default = 0.75
+            A multiplier to max_memory_usage_ratio to get the max_memory_usage_warning_ratio
+            If expected memory usage is >max_memory_usage_error_ratio, raise NotEnoughMemoryError
+        mem_size_threshold : int, default = None
+            If not None, skips checking available memory if the expected model size is less than `mem_size_threshold` bytes.
+            This is used to speed-up training by avoiding the check in cases where the machine almost certainly has sufficient memory.
+        **kwargs : dict,
+            Fit time kwargs, including X, y, X_val, and y_val.
+            Can be used to customize estimation of memory usage.
+        """
         max_memory_usage_ratio = self.params_aux['max_memory_usage_ratio']
+        if max_memory_usage_ratio is None:
+            return  # Skip memory check
+
         approx_mem_size_req = self.estimate_memory_usage(**kwargs)
+        if mem_size_threshold is not None and approx_mem_size_req < (mem_size_threshold * min(max_memory_usage_ratio, 1)):
+            return  # Model is smaller than the min threshold to check available mem
+
         available_mem = ResourceManager.get_available_virtual_mem()
         ratio = approx_mem_size_req / available_mem
-        if ratio > (0.9 * max_memory_usage_ratio):
-            logger.warning('\tWarning: Not enough memory to safely train model, roughly requires: %s GB, but only %s GB is available...' % (round(approx_mem_size_req / 1e9, 3), round(available_mem / 1e9, 3)))
+        min_error_memory_ratio = ratio / mem_error_threshold
+        min_warning_memory_ratio = ratio / mem_warning_threshold
+        max_memory_usage_error_ratio = mem_error_threshold * max_memory_usage_ratio
+        max_memory_usage_warning_ratio = mem_warning_threshold * max_memory_usage_ratio
+
+        log_ag_args_fit_example = '`predictor.fit(..., ag_args_fit={"ag.max_memory_usage_ratio": VALUE})`'
+        log_ag_args_fit_example = f'\n\t\tTo set the same value for all models, do the following when calling predictor.fit: {log_ag_args_fit_example}'
+
+        log_user_guideline = f'Estimated to require {round(approx_mem_size_req / 1e9, 3)} GB ' \
+                             f'out of {round(available_mem / 1e9, 3)} GB available memory ({round(min_error_memory_ratio*100, 3)}%)... ' \
+                             f'({round(max_memory_usage_error_ratio*100, 3)}% of avail memory is the max safe size)'
+        if min_error_memory_ratio > max_memory_usage_error_ratio:
+            log_user_guideline += f'\n\tTo force training the model, specify the model hyperparameter "ag.max_memory_usage_ratio" to a larger value ' \
+                                  f'(currently {max_memory_usage_ratio}, set to >={round(min_error_memory_ratio + 0.05, 2)} to avoid the error)' \
```

Comment:
`round` is not required - formatter supports that out-of-the box:
```python
f'(currently {max_memory_usage_ratio}, set to >={min_warning_memory_ratio + 0.05:.2f} to avoid the warning)'
```

Reply:
good catch, updated